### PR TITLE
Source refactoring

### PR DESCRIPTION
**Pull request by simleo**
Refactors C++ source into multiple modules.

### CODE REVIEW

File: src/errors.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+#ifndef OME_FILES_PY_ERRORS_H
```
Comment (rleigh-codelibre) [inline src/errors.h]: Just as a note, you can use

```
#pragma once
```

in place of the traditional three line include guards.  I noticed a few weeks back that it's supported by all the common compilers we target (gcc, clang, msvc), and I tested this with the ome-files-performance repo.  Should be safer and simpler should you wish to try it out (I'm tempted to switch over all the C++ code should it be acceptable).

File: src/omefiles.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ * #L%
+ */
+
+#include <Python.h>
+
+#include "ometiffreader.h"
+#include "errors.h"
+
+
+PyObject *OMEFilesPyError;
```
Comment (rleigh-codelibre) [inline src/omefiles.cpp]: Is this not redefining what you already declared in `errors.h`?

Comment (rleigh-codelibre) [inline src/omefiles.cpp]: Sorry, just realised this was the implementation, ignore this.

**Comment by rleigh-codelibre** [comment]
The changes look fine, just one note added (doesn't need any changes unless you want to).

The only minor comment would be why `errors.h` is a plural and not just `error.h`?  Are more errors going to be added here?

Green in https://ci.openmicroscopy.org/view/Files/job/OME-FILES-CPP-DEV-merge-superbuild/BUILD_TYPE=Release,node=cowfish/521/consoleText (the other builds are still running).

**Comment by simleo** [comment]
Yes, the idea was that at some point we might want to add more errors, corresponding to the different exceptions thrown by the C++ side.

**Pull request merged**